worksheet.rb

```diff
@@ -23,12 +39,159 @@
 # into this data structure, such as "DR0004"
 # and "3rd Feb 2016" and "RD0022"
 
+drivers = [
+  {
+    id: 'DR0004',
+    trips: [
+      {
+        date: '3rd Feb 2016',
+        cost: 5,
+        rider_id: 'RD0022',
+        rating: 5
+      },
+      {
+        date: "4th Feb 2016",
+        cost: 10,
+        rider_id: "RD0022",
+        rating: 4
+      },
+      {
+        date: "5th Feb 2016",
+        cost: 20,
+        rider_id: "RD0073",
+        rating: 5
+      }
+    ]
+  },
+  {
+    id: 'DR0001',
+    trips: [
+      {
+        date: "3rd Feb 2016",
+        cost: 10,
+        rider_id: "RD0003",
+        rating: 3
+      },
+      {
+        date: "3rd Feb 2016",
+        cost: 30,
+        rider_id: "RD0015",
+        rating: 4
+      },
+      {
+        date: "5th Feb 2016",
+        cost: 45,
+        rider_id: "RD0003",
+        rating: 2
+      }
+    ]
+  },
+  {
+    id: 'DR0002',
+    trips: [
+      {
+        date: "3rd Feb 2016",
+        cost: 25,
+        rider_id: "RD0073",
+        rating: 5
+      },
+      {
+        date: "4th Feb 2016",
+        cost: 15,
+        rider_id: "RD0013",
+        rating: 1
+      },
+      {
+        date: "5th Feb 2016",
+        cost: 45,
+        rider_id: "RD0003",
+        rating: 2
+      }
+    ]
+  },
+  {
+    id: 'DR0003',
+    trips: [
+      {
+        date: "4th Feb 2016",
+        cost: 5,
+        rider_id: "RD0066",
+        rating: 5
+      },
+      {
+        date: "5th Feb 2016",
+        cost: 50,
+        rider_id: "RD0003",
+        rating: 2
+      },
+      {
+        date: "5th Feb 2016",
+        cost: 35,
+        rider_id: "RD0066",
+        rating: 6
+      }
+    ]
+  }
+]
+
 ########################################################
+def total_rides(driver)
+  return driver[:trips].length
+end
+
+def driver_income(driver)
+  return driver[:trips].sum{ |trip| trip[:cost] }
+end
+
+def driver_average_rating(driver)
+  driver_rating_sum = driver[:trips].sum{ |trip| trip[:rating] }
+  return driver_rating_sum.to_f / driver[:trips].length
```

Comment:
You wrote a great helper function `total_rides`. Consider using it here!

```suggestion
  return driver_rating_sum.to_f / total_rides(driver)
```